packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,208 @@
+      const searchboxClearedEvent = new CustomEvent('sfx::search_box_cleared');
+
+      searchbox.emitSearchBoxClearedEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxClearedEvent);
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the value property in response to data received', () => {
+      const detail = 'inputText';
+      stub(searchbox, 'getInputElement').returns(
+        html`
+          <input type="text" data-sfx-ref="searchInput" placeholder="Type your search" />
+        `
+      );
+      searchbox.updateText({ detail });
+      expect(searchbox.value).to.equal(detail);
+    });
+  });
+
+  describe('handleKeydown', () => {
+    it('should invoke the emitSearchEvent function if enter is pressed and value property length is greater than 0', () => {
+      const emitSearchStub = stub(searchbox, 'emitSearchEvent');
+      searchbox.value = 'hello';
+
+      searchbox.handleKeydown({ keyCode: KEY_CODES.ENTER });
+
+      expect(emitSearchStub).to.have.been.called;
+    });
+  });
+
+  describe('handleChange', () => {
+    it('should invoke the updateSearchTermValue function', () => {
+      const updateSearchTermValueStub = stub(searchbox, 'updateSearchTermValue');
+      const target = { value: 'dee' };
+
+      searchbox.handleChange({ target });
+
+      expect(updateSearchTermValueStub).to.have.been.calledWith('dee');
+    });
+
+    it('should dispatch a search box change event', () => {
+      const target = { value: 'dee' };
+      const searchboxChangeEvent = new CustomEvent('sfx::on_search_box_change', {
+        detail: target,
+        bubbles: true
+      });
+      stub(searchbox, 'updateSearchTermValue');
+
+      searchbox.handleChange({ target });
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxChangeEvent);
+    });
+  });
+
+  describe('updateSearchTermValue', () => {
+    it('should set the search term property to the input value', () => {
+      searchbox.updateSearchTermValue('catfood');
+
+      expect(searchbox.value).to.equal('catfood');
+    });
+  });
+
+  describe('clearSearch', () => {
+    it('should set the search term property to an empty string', () => {
+      stub(searchbox, 'getInputElement').returns(
+        html`
+          <input type="text" id="searchInput" placeholder="Type your search" />
+        `
+      );
+      stub(searchbox, 'emitSearchBoxClearedEvent');
+
+      searchbox.clearSearch();
+
+      expect(searchbox.value).to.equal('');
+    });
+
+    it('should invoke the emitSearchBoxClearedEvent', () => {
+      const emitSearchBoxSpy = spy(searchbox, 'emitSearchBoxClearedEvent');
+      stub(searchbox, 'getInputElement').returns(
+        html`
+          <input type="text" id="searchInput" placeholder="Type your search" />
+        `
+      );
+      
+      searchbox.clearSearch();
+
+      expect(emitSearchBoxSpy).to.have.been.called;
+    });
+  });
```

Comment:
This is missing the test for setting the value of the input box.

Reply:
Rolled this into the first test, please review.